### PR TITLE
Added info about "props:card" option

### PR DESCRIPTION
**Pull request by aaronrothschild**
- Added 'card' to table of webhook parameters
- Added an example that contains a meta-tag and a screenshot of how it looks

### CODE REVIEW

File: site/content/integrate/incoming-webhooks/_index.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ Incoming webhooks support more than just the `text` field. Here is a full list o
 | attachments | [Message attachments](https://docs.mattermost.com/developer/message-attachments.html) used for richer formatting options. | If `text` is not set, yes |
 | type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with "custom\_". | No |
 | props | Sets the post `props`, a JSON property bag for storing extra or meta data on the post.<br> Mainly used by other integrations accessing posts through the REST API.<br> The following keys are reserved: "from\_webhook", "override\_username", "override\_icon\_url", "override\_icon\_emoji", "webhook\_display\_name" and "attachments". | No |
+| props | Props `card` allows for extra information (markdown formatted text) to be sent to Mattermost that will only be displayed in the RHS panel after a user clicks on an 'info' icon displayed alongside the post.  The info icon cannot be customized and is only rendered visible to the user if there is `card` data passed into the message. This is only available in v5.14+. | No |
```
Comment (jasonblais) [inline site/content/integrate/incoming-webhooks/_index.md]: Should we incorporate it as an additional description to the `props` parameter on line 70?

File: site/content/integrate/incoming-webhooks/_index.md [hunk 2]
```diff
@@ -85,6 +86,26 @@ Content-Length: 630
 }
 ```
 
+An example request that would display additional info in the RHS panel, by passing markdown text into `props` as a `card` object would look like this:
```
Comment (jasonblais) [inline site/content/integrate/incoming-webhooks/_index.md]:
```suggestion
An example request displaying additional data in the right-hand side panel, by passing Markdown text into `props` as a `card` object would look like this:
```

File: site/content/integrate/incoming-webhooks/_index.md [hunk 3]
```diff
@@ -85,6 +86,26 @@ Content-Length: 630
...
+POST /hooks/xxx-generatedkey-xxx HTTP/1.1
+Host: your-mattermost-server
+Content-Type: application/json
+
+
+{
+  "channel": "town-square",
+  "username": "Winning-bot",
+  "text": "#### We won a new deal!"
+  "props": "{"card": "Salesforce Opportunity Information:\n\n [Opportunity Name](http://salesforce.com/OPPORTUNITY_ID)\n\n-Salesperson: **Bob McKnight** \n\n Amount: **$300,020.00**"}
```
Comment (jasonblais) [inline site/content/integrate/incoming-webhooks/_index.md]:
```suggestion
  "props": "{"card": "Salesforce Opportunity Information:\n\n [Opportunity Name](http://salesforce.com/OPPORTUNITY_ID)\n\n-Salesperson: **Bob McKnight** \n\n Amount: **$300,020.00**"}"
```

File: site/content/integrate/incoming-webhooks/_index.md [hunk 4]
```diff
@@ -85,6 +86,26 @@ Content-Length: 630
...
+}
+```
+
+When there is `props:card` data passed in via the webhook, the rendered message will display a small icon that expands the RHS panel to display the markdown that was passed in with the webhook:
```
Comment (jasonblais) [inline site/content/integrate/incoming-webhooks/_index.md]:
```suggestion
When there is `props:card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in `props:card`:
```

File: site/content/integrate/incoming-webhooks/_index.md (hunk 1 above)
Comment (jasonblais) [reply site/content/integrate/incoming-webhooks/_index.md]: Also, would we want to mention this is not supported on mobile?

Comment (hanzei) [reply site/content/integrate/incoming-webhooks/_index.md]: Is this just available for webhooks or for posts in general?

Comment (aaronrothschild) [reply site/content/integrate/incoming-webhooks/_index.md]: @hanzei Available for posts in General.  I'm updating the Zapier connector for example which is posting via the Rest API and has posted "card" data via the API (not webhook).  I guess we should also document it somewhere in the API docs too?

File: site/content/integrate/incoming-webhooks/_index.md [hunk 5]
```diff
@@ -68,6 +68,7 @@ Incoming webhooks support more than just the `text` field. Here is a full list o
 | attachments | [Message attachments](https://docs.mattermost.com/developer/message-attachments.html) used for richer formatting options. | If `text` is not set, yes |
 | type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with "custom\_". | No |
 | props | Sets the post `props`, a JSON property bag for storing extra or meta data on the post.<br> Mainly used by other integrations accessing posts through the REST API.<br> The following keys are reserved: "from\_webhook", "override\_username", "override\_icon\_url", "override\_icon\_emoji", "webhook\_display\_name" and "attachments". | No |
```
Comment (jespino) [inline site/content/integrate/incoming-webhooks/_index.md]: I miss the `card` field as reserved word here.

File: site/content/integrate/incoming-webhooks/_index.md [hunk 6]
```diff
@@ -85,6 +86,26 @@ Content-Length: 630
 }
 ```
 
+An example request displaying additional data in the right-hand side panel, by passing Markdown text into `props` as a `card` object would look like this:
```
Comment (jespino) [inline site/content/integrate/incoming-webhooks/_index.md]: I think is not clear enough here. I would say something like:

```suggestion
An example request displaying additional data in the right-hand side panel, by passing Markdown text into the `card` field of the `props` object would look like this:
```

File: site/content/integrate/incoming-webhooks/_index.md [hunk 7]
```diff
@@ -85,6 +86,26 @@ Content-Length: 630
...
+
+{
+  "channel": "town-square",
+  "username": "Winning-bot",
+  "text": "#### We won a new deal!"
+  "props": "{"card": "Salesforce Opportunity Information:\n\n [Opportunity Name](http://salesforce.com/OPPORTUNITY_ID)\n\n-Salesperson: **Bob McKnight** \n\n Amount: **$300,020.00**"}"
+}
+```
+
+When there is `props:card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in `props:card`:
```
Comment (jespino) [inline site/content/integrate/incoming-webhooks/_index.md]:
```suggestion
When there is `props` object with a `card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in the `card` property:
```

File: site/content/integrate/incoming-webhooks/_index.md (hunk 1 above)
Comment (hanzei) [reply site/content/integrate/incoming-webhooks/_index.md]: I'm not sure what you mean with "API docs". https://api.mattermost.com/?

I would like (3/5) to see a new page in https://developers.mattermost.com/integrate/ where all post parameters are described. What do you think about this?

Comment (aaronrothschild) [reply site/content/integrate/incoming-webhooks/_index.md]: @hanzei - Yes, the API docs there.  You can now pass in "Props" with "card" to an HTTP endpoint.  How would you know that by looking at this docs page? 
image

Curious - the page you'd like to see, would it be called "Global Post Parameters" or something like that?  Is there anywhere these Post-Params can be used *aside* from Incoming Webhooks and the REST API?   The reason I ask, is when looking through the dev.docs site, if I went to the "Integrate" area - you would likely either go to the Webhooks page (which we can ensure document all the params further) OR to the api.mattermost.com site which would then have the HTTP API Post documentation (and currently lacks any information about "card",  etc.).  Why would we expect someone to go to a separate page to find out about further options about the /api/v4/post endpoint?

File: site/content/integrate/incoming-webhooks/_index.md (hunk 5 above)
Comment (aaronrothschild) [reply site/content/integrate/incoming-webhooks/_index.md]: I added the 'card' field to reserved words.

File: site/content/integrate/incoming-webhooks/_index.md (hunk 1 above)
Comment (jasonblais) [reply site/content/integrate/incoming-webhooks/_index.md]: On my original question, there are two 'props' parameters in the table, but should there be just one (both within the same `props` parameter)?

image

Comment (aaronrothschild) [reply site/content/integrate/incoming-webhooks/_index.md]: Will remove second box and merge

**Comment by aaronrothschild** [comment]
@amyblais Why were the labels added back, Is there someone that needs to review this again?

**Comment by jasonblais** [comment]
@aaronrothschild I think Amy added the dev review label since there was feedback from Hanzei previously

**Comment by amyblais** [comment]
@aaronrothschild I thought there was an unresolved question on this thread https://github.com/mattermost/mattermost-developer-documentation/pull/355#discussion_r321923839. Feel free to merge if this is ready.

**Pull request merged**